File: demo-vk-rest-scraping.java/src/main/java/demo/reader/ObjectsListRepeatableReader.java
package demo.reader;

import com.google.api.client.util.ArrayMap;
import demo.model.ObjectsList;

import java.util.List;
import java.util.function.Function;
import java.util.stream.Collectors;

public class ObjectsListRepeatableReader extends RepeatableReader<ObjectsList, Object> {

    public ObjectsListRepeatableReader() {
        super(ObjectsList.class);
    }

    public <T> List<T> read(String urlBody, Function<? super ArrayMap, T> mapper) {
        return read(urlBody)
                .response()
                .stream()
                .skip(1)
                .map(object -> (ArrayMap) object)
                .map(mapper)
                .collect(Collectors.toList());
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/reader/ObjectsListRepeatableReader.kt
package demo.reader

import com.google.api.client.util.ArrayMap
import demo.model.ObjectsList

class ObjectsListRepeatableReader : RepeatableReader<ObjectsList, Any>(ObjectsList::class.java) {

    fun <T> read(urlBody: String, mapper: (ArrayMap<*, *>) -> T): List<T> {
        val list = read(urlBody)
                .response()!!

        return list.subList(1, list.size)
                .map({ `object` -> `object` as ArrayMap<*, *> })
                .map(mapper)
    }
}
